library/core/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -355,6 +357,8 @@
   @C.BufferFlags
   private int blockFlags;
 
+  private int lastReadResult = Extractor.RESULT_CONTINUE;
```

Comment:
Maybe just store private boolean readEndOfInput, set to true if continueReading is false in read()?

Reply:
Yeah... that is much easier to deal with that way.  I think part of that is a holder from a more needlessly complex implementation.